src/pretalx/submission/models/submission.py

```diff
@@ -288,39 +299,70 @@ def _set_state(self, new_state, force=False, person=None):
                 )
             )
 
-    def make_submitted(self, person=None, force=False, orga=False):
-        self._set_state(SubmissionStates.SUBMITTED, force, person=person)
+    def update_talk_slots(self, slot_count_old=None):
         from pretalx.schedule.models import TalkSlot
+        # logging.debug('update_talk_slots')
+        # debug: print all now available Talks
+        # qs_temp = TalkSlot.objects.filter(
+        #     submission=self,
+        #     schedule=self.event.wip_schedule,
+        # )
+        # logging.debug('TalkSlots old: {}'.format(
+        #     [i.slot_index for i in qs_temp]))
+        # logging.debug('self.state: {}'.format(self.state))
+        if self.state == SubmissionStates.ACCEPTED or \
+                self.state == SubmissionStates.CONFIRMED:
+            # logging.debug('state is ACCEPTED or CONFIRMED')
+            if slot_count_old is not None:
+                # logging.debug(
+                #     'slot_count_old: {}'.format(slot_count_old))
+                # logging.debug(
+                #     'self.slot_count {}'.format(self.slot_count))
+                range_to_delete = list(range(
+                    self.slot_count, slot_count_old))
+                # logging.debug('range_to_delete: {}'.format(range_to_delete))
+                qs_delete = TalkSlot.objects.filter(
+                    submission=self,
+                    schedule=self.event.wip_schedule,
+                    slot_index__in=range_to_delete,
+                )
+                # logging.debug('qs_delete: {}'.format(
+                #     [i.slot_index for i in qs_delete]))
+                qs_delete.delete()
+            # logging.debug('TalkSlot update_or_create')
+            for index in range(self.slot_count):
+                TalkSlot.objects.update_or_create(
+                    submission=self,
+                    schedule=self.event.wip_schedule,
+                    defaults={'is_visible': True},
```

Comment:
Visibility should be `True` if confirmed, `False` if accepted, and not just default to True.

Reply:
hm ok - will change this.

i just copied that over - from the original 
confirm [pretalx/submission/models/submission.py#L308](https://github.com/pretalx/pretalx/blob/master/src/pretalx/submission/models/submission.py#L308)
accept [pretalx/submission/models/submission.py#L321](https://github.com/pretalx/pretalx/blob/master/src/pretalx/submission/models/submission.py#L321)
functions..

Reply:
i think i have found a hint why this was 'alwas true'
- the is_visible field will be set by the schedule freeze function.
so it is not relevant how it is initialized.
i have allowed me the fredome to define False as default value for this field and remove the 'update visibility' lines in my update function...
please - if i have overlooked something regarding this let me know!